packages/coil-extension/src/background/services/SiteToken.ts

```diff
@@ -18,10 +18,34 @@ import { timeoutRejecting } from '../../util/timeout'
 export class SiteToken {
   constructor(
     @inject(tokens.CoilDomain)
-    private coilDomain: string
+    private coilDomain: string,
+    @inject(tokens.WextApi)
+    private api = chrome
   ) {}
 
   async retrieve(path = '/handler.html'): Promise<string | null> {
+    return new Promise(resolve => {
+      this.api.tabs.create(
+        {
+          active: false,
```

Comment:
@sharafian 

This actually isn't too bad UX. It opens a tab without switching your focus at all, then closes it once it has retrieved the token. If [restoring handler.html](https://github.com/coilhq/coil/pull/4865) doesn't resolve the issue on [Firefox](https://github.com/coilhq/web-monetization-projects/issues/1941#issuecomment-849288125) perhaps we could live with this?

Reply:
For the record, the handler fix fixed the freeze on FF